src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +687,45 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #unlock' do
+    before do
+      login user
+    end
+
+    context "with a project that is locked" do
+      before do
+        user.home_project.flags.create(flag: 'lock', status: 'enable')
+      end
+
+      it 'successfully unlocks the project' do
+        post :unlock, project: user.home_project
+        is_expected.to redirect_to(action: :show, project: user.home_project)
+        expect(flash[:notice]).to eq('Successfully unlocked project')
+      end
+
+      context "with a project that has maintenance release requests" do
+        let!(:bs_request) { create(:bs_request, type: 'maintenance_release', source_project: user.home_project.name) }
+
+        before do
+          user.home_project.update(kind: 'maintenance_incident')
```

Comment:
And it may be a problem, as we assume that the relation is present and can be used but maybe it doesn't exist.